### ISSUE TITLE
R Parser Incorrectly Identifies Number with Bracket as Parse Error

### ISSUE THREAD

**Issue by billdenney** [rstudio/rstudio#8941]
### System details

    RStudio Edition : Desktop
    RStudio Version : 1.4.1103
    OS Version      : Windows 10
    R Version       : 4.0.3

### Steps to reproduce the problem

Paste the following code into an R script and save it.

```r
1[FALSE]
1[TRUE]
"A"[FALSE]
"A"[TRUE]
```

### Describe the problem in detail

The parser identifies the first two rows with a number followed by an indexing bracket as having an error of "Unexpected token '['"

![image](https://user-images.githubusercontent.com/10359931/107431352-54ab2980-6af4-11eb-9e81-8467ea94ecd8.png)

### Describe the behavior you expected

No parse error (it is valid R).

- [x] I have read the guide for [submitting good bug reports](https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports).
- [x] I have installed the latest version of RStudio, and confirmed that the issue still persists.
- [ ] If I am reporting a RStudio crash, I have included a [diagnostics report](https://support.rstudio.com/hc/en-us/articles/200321257-Running-a-Diagnostics-Report).
- [x] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by ronblum** [comment]
@billdenney Thank you for filing this issue! I can reproduce it on:
- RStudio Desktop 1.4.1554 on MacOS 11.3
- RStudio Server 1.4.1554 on Red Hat 8.3 via Chrome on MacOS

And this code is, as you've stated, is valid in  iR:
```
> 1[FALSE]
numeric(0)
> 1[TRUE]
[1] 1
> "A"[FALSE]
character(0)
> "A"[TRUE]
[1] "A"
```

**Comment by billdenney** [comment]
This appears to have been resolved by Rstudio version 1.4.1717.
